Specifications/Aggregates/for_AggregateRoot/given/committed_events_and_two_aggregate_roots.cs

```diff
@@ -29,39 +30,54 @@ public abstract class committed_events_and_two_aggregate_roots : two_aggregate_r
+    public static async IAsyncEnumerable<CommittedAggregateEvents> build_committed_events(EventSourceId eventSource, AggregateRootId aggregateRootId, ExecutionContext executionContext)
     {
-        var events = new List<CommittedAggregateEvent>()
+        var events = Enumerable.Range(0, 3)
+            .Select(i =>
+            {
+                object @event = i switch
+                {
+                    0 => first_event,
+                    1 => second_event,
+                    2 => third_event
+                };
+
+                return build_committed_event(
+                    eventSource,
+                    aggregateRootId,
+                    (ulong) i,
+                    (ulong) i,
+                    @event,
+                    event_types.GetFor(@event.GetType()),
+                    false,
+                    executionContext);
+            });
+        yield return new CommittedAggregateEvents(eventSource, aggregateRootId, 3, events.ToList());
+    }
+
+    public static async IAsyncEnumerable<CommittedAggregateEvents> build_committed_events_batches(EventSourceId eventSource, AggregateRootId aggregateRootId, ExecutionContext executionContext)
+    {
+        foreach (ulong i in Enumerable.Range(0, 3))
         {
-            build_committed_event(
-                eventSource,
-                aggregateRootId,
-                0,
-                0,
-                first_event,
-                event_types.GetFor(first_event.GetType()),
-                false,
-                executionContext),
-            build_committed_event(
-                eventSource,
-                aggregateRootId,
-                1,
-                1,
-                second_event,
-                event_types.GetFor(second_event.GetType()),
-                false,
-                executionContext),
-            build_committed_event(
-                eventSource,
-                aggregateRootId,
-                2,
-                2,
-                third_event,
-                event_types.GetFor(third_event.GetType()),
-                false,
-                executionContext),
-        };
-        return new CommittedAggregateEvents(eventSource, aggregateRootId, events);
+            object @event = i switch
+            {
+                0 => first_event,
+                1 => second_event,
+                2 => third_event
+            };
+            yield return new CommittedAggregateEvents(eventSource, aggregateRootId, 3, new []
+            {
+                build_committed_event(
+                    eventSource,
+                    aggregateRootId,
+                    i,
+                    i,
+                    @event,
+                    event_types.GetFor(@event.GetType()),
+                    false,
+                    executionContext)
+            });
+        }
```

Comment:
Just for completeness of testing, it could be nice if there was one batch with multiple events in it? Just to make sure we don't like only process the first one in the batch...

Reply:
Yes we should. The specs were just a little pain that's why this was a bit sloppy